tokio/src/sync/semaphore.rs

```diff
@@ -210,6 +210,55 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
+/// ## Ensure Tests with Mutual Dependencies Run Sequentially
+///
+/// By default, Rust runs tests concurrently. However, in some scenarios, concurrent execution might lead to test interference.
+/// For example, multiple tests might communicate with the same database, necessitating sequential execution to prevent data conflicts.
+///
+/// Consider the following scenario:
+/// 1. `test_insert`: Inserts a key-value pair into the database, then retrieves the value using the same key to verify the insertion.
+/// 2. `test_update`: After insertion, this test updates the key's value to a new one and verifies if the value has been accurately updated.
+/// 3. `test_others`: This is a non-interfering test that can run concurrently with other tests.
+///
+/// For this example, it's essential that `test_insert` and `test_update` run sequentially, though their execution order is inconsequential.
+/// Leveraging a semaphore with a single permit elegantly addresses this challenge.
+///
+/// ```
+/// #[cfg(test)]
+/// mod tests {
+///    use tokio::sync::Semaphore;
+///    // only one permit can be acquired
+///    static PERMIT: Semaphore = Semaphore::const_new(1);
+///    // initilization of database
```

Comment:
The other examples use proper sentences as comments. I'd like to be consistent.

Reply:
done.